vm/src/stdlib/collections.rs

```diff
@@ -205,10 +205,24 @@ mod _collections {
         }
 
         #[pymethod]
-        fn extendleft(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
-            for elem in iter.iter(vm)? {
-                self.appendleft(elem?);
+        fn extendleft(&self, iter: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            let max_len = self.maxlen;
+            let mut elements: Vec<PyObjectRef> = vm.extract_elements(&iter)?;
+            elements.reverse();
+            let elements_len = elements.len();
+
+            if let Some(max_len) = max_len {
+                if max_len > elements.len() {
+                    let mut deque = self.borrow_deque_mut();
+                    let drain_until = deque.len().saturating_sub(max_len - elements.len());
+                    deque.drain(..drain_until);
```

Comment:
This should be a `truncate` call. Recall, its the opposite of `extend`, so you need to trim of elements from the end. A `truncate_until` similar to `drain_until` needs to be calculated that denotes how many elements must be removed *from the front* to make room for `elements`.